sql/src/main/antlr/OpenSearchSQLParser.g4

```diff
@@ -310,6 +310,23 @@ functionCall
     | highlightFunction                                             #highlightFunctionCall
     | positionFunction                                              #positionFunctionCall
     | getFormatFunction                                             #getFormatFunctionCall
+    | timestampAddFunction                                          #timestampAddFunctionCall
+    ;
+
+timestampAddFunction
+    : TIMESTAMPADD LR_BRACKET timstampAddIntervalType COMMA length=functionArg COMMA timestampExpr=functionArg RR_BRACKET
+    ;
+
+timstampAddIntervalType
+    : MICROSECOND
```

Comment:
Can we re-use common parts with `extract` function?
Something like:
```
timestampAddFunction
    : ... simpleDateTimePart ... ;

extractFunction
    : ... (simpleDateTimePart | complexDateTimePart) ... ;

```
I think it is used somewhere else too

Reply:
Reworked in https://github.com/Bit-Quill/opensearch-project-sql/pull/246/commits/e2fdcd1779f4d009b44ab138c3d22fa38e7af021.
It will also eventually get used int `timestampdiff`.